packages/editor/src/functions/getIntersectingNode.ts

```diff
@@ -1,27 +1,37 @@
 import { Engine } from '@xrengine/engine/src/ecs/classes/Engine'
+import { EntityTreeNode } from '@xrengine/engine/src/ecs/classes/EntityTree'
+import { useWorld } from '@xrengine/engine/src/ecs/functions/SystemHooks'
+import { Object3DWithEntity } from '@xrengine/engine/src/scene/components/Object3DComponent'
 import { Camera, Intersection, Object3D, Raycaster, Vector2 } from 'three'
 
-type RaycastIntersectionNode = Intersection<Object3D> & {
-  node: Object3D
+type RaycastIntersectionNode = Intersection<Object3DWithEntity> & {
+  obj3d: Object3DWithEntity
+  node?: EntityTreeNode
 }
 
-export function getIntersectingNode(results: Intersection<Object3D>[], scene: Object3D): RaycastIntersectionNode {
-  if (results.length > 0) {
-    for (const result of results) {
-      let curObject = result.object as any
-      while (curObject) {
-        if (curObject.isNode) {
-          break
-        }
-        curObject = curObject.parent
-      }
-      if (curObject && curObject !== scene && !curObject.ignoreRaycast) {
-        ;(result as RaycastIntersectionNode).node = curObject
-        return result as RaycastIntersectionNode
-      }
+function getParentEntity(obj: Object3DWithEntity): Object3DWithEntity {
```

Comment:
could we do an Object3D.traverseParents to avoid the while loop?

Reply:
In `traverseParents` breaking the loop and early exit is not possible.